### DISCUSSION TITLE
✅ How to test Sign-In on local machine? Got error on both Desktop and Web app

### DISCUSSION THREAD

**Discussion by taurenshaman** [idena-network/idena-desktop#764] [Show and tell]
Hello, I'm wanna add Indena Sign-In on my website. But I got errors on both Desktop app and Web app.

I'm testing my website on `https://localhost:44385`

### 1. While using desktop app
The generated hyperlink is:
```
// desktop
dna://signin/v1?token=f7c014f4-eaff-4915-a87c-94f213f39290&callback_url=https%3A%2F%2Flocalhost:44385&nonce_endpoint=https%3A%2F%2Flocalhost:44385%2Fidena%2Fstart-session&authentication_endpoint=https%3A%2F%2Flocalhost:44385%2Fidena%2Fauthenticate&favicon_url=https%3A%2F%2Flocalhost:44385%2Ffavicon.ico
```

![image](https://user-images.githubusercontent.com/1308706/157809546-6c3d367b-6a36-4f09-88b2-220d78a07600.png)

After click `confirm`, I got error:
`String reference not set to an instance of a String. (Parameter 's')`

![image](https://user-images.githubusercontent.com/1308706/157809605-d58a3b59-181d-4ac1-82a6-91913fe423f7.png)

### 2. While using web app
The generated hyperlink is:
```
// web
https://app.idena.io/dna/signin?token=ecf04978-b851-4c86-a359-5d3ffcc1558c&callback_url=https://localhost:44385&nonce_endpoint=https://localhost:44385/idena/start-session&authentication_endpoint=https://localhost:44385/idena/authenticate&favicon_url=https://localhost:44385/favicon.ico
```
After click `confirm`, I got error:
`connect ECONNREFUSED 127.0.0.1:44385`

**Comment by midenaio** [comment]
Hello @taurenshaman, 

> After click confirm, I got error:
> connect ECONNREFUSED 127.0.0.1:44385

> I'm testing my website on https://localhost:44385

Have you tried `http://localhost:44385` instead of `httpS://localhost:44385`?

> String reference not set to an instance of a String. (Parameter 's')

It seems the error is being returned by your auth server.

**Reply by taurenshaman** [reply]
Thank you very much. It's my fault. I fogot to check the data in Request body content.

**Comment by taurenshaman** [comment]
Anyone who are interesting to use C# can have a look at [my code](https://github.com/rune-box/connect/blob/main/src/Connect/Controllers/IdenaController.cs) .
The simple test is OK:
![idena-auth-ok](https://user-images.githubusercontent.com/1308706/158096809-ccddadcc-38cb-402a-95f4-54a2ff032b67.png)
